test/internal/scorecard.go

```diff
@@ -79,3 +79,17 @@ func (tc TestContext) AddScorecardCustomPatchFile() error {
 	}
 	return nil
 }
+
+func (tc TestContext) AddScorecardDevImage() {
+	// Replace the scorecard dev image in config files.
+	ReplaceRegexInFile(
+		filepath.Join(tc.Dir, "config", "scorecard", "patches", "basic.config.yaml"),
+		"quay.io/operator-framework/scorecard-test:.*",
+		"quay.io/operator-framework/scorecard-test:dev",
+	)
+	ReplaceRegexInFile(
+		filepath.Join(tc.Dir, "config", "scorecard", "patches", "olm.config.yaml"),
+		"quay.io/operator-framework/scorecard-test:.*",
+		"quay.io/operator-framework/scorecard-test:dev",
+	)
+}
```

Comment:
IMO we could to do:

```go
By("replacing scorecard-test image to use the dev tag")
const scorecardImage :="quay.io/operator-framework/scorecard-test:master"
const scorecardImageReplace :="quay.io/operator-framework/scorecard-test:dev"
err = testutils.ReplaceInAllFilesFromTree(filepath.Join(tc.Dir, "bundle"), scorecardImage,scorecardImageReplace)
Expect(err).NotTo(HaveOccurred())
err = testutils.ReplaceInAllFilesFromTree(filepath.Join(tc.Dir, "config", "scorecard"), scorecardImage, scorecardImageReplace)
Expect(err).NotTo(HaveOccurred())
```

```go
// ReplaceInAllFiles replaces all instances of old with new in the files of a directory and its subdirectories.
func ReplaceInAllFilesFromTree(projectPath, old, new string) error {
	err := filepath.Walk(projectPath,
		func(path string, info os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			if !info.IsDir() {
				ReplaceInFile(path, old, new)
			}
			return nil
		})
	if err != nil {
		return err
	}
	return nil
}
```


Motivation: The above code will ensure that we are always replacing ALL places. Note that in the future we will build the bundle **before** in the samples and then, we will do this replace just for the e2e tests.  

Also, note that it is very easy we add new scaffolds and just forget that we need to replace the images. So, I think it is better we add the func that will replace ALL in a tree as suggested above for we do not end up by mistake in this situation. 

However, it can definitely be a follow-up as well such as: 
- PR to centralize
- PR to improve

Reply:
Thanks @camilamacedo86 . We should address this as well. Should this I handle in this PR or I should create a different ticket explaining why we are doing it and add a follow up PR?

Reply:
think maybe it will be easy in a follow-up, no worry.